Spigot-Server-Patches/0600-Allow-toggling-special-MobSpawners-per-world.patch

```diff
@@ -0,0 +1,75 @@
++    private void mobSpawnerSettings() {
++        // values of "default" or null will default to true only if the world environment is normal (aka overworld)
++        Predicate<Boolean> predicate = (bool) -> (bool != null && bool) || (bool == null && environment == Environment.NORMAL);
++        catSpawning = getBoolean("mob-spawning.village-cats", predicate);
++        patrolSpawning = getBoolean("mob-spawning.raid-patrols", predicate);
++        phantomSpawning = getBoolean("mob-spawning.phantoms", predicate);
++        villagerTraderSpawning = getBoolean("mob-spawning.wandering-traders", predicate);
++        villageSiegeSpawning = getBoolean("mob-spawning.village-sieges", predicate);
++    }
++
+     public int containerUpdateTickRate;
+     private void containerUpdateTickRate() {
+         containerUpdateTickRate = getInt("container-update-tick-rate", 1);
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index d0ffe5790e98b65cd6035ad529e7da8172de9029..9f90cf4c9da8b870eb6cbc468356d5a82035b2c6 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -402,7 +402,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
+             boolean flag = generatorsettings.isDebugWorld();
+             long i = generatorsettings.getSeed();
+             long j = BiomeManager.a(i);
+-            List<MobSpawner> list = ImmutableList.of(new MobSpawnerPhantom(), new MobSpawnerPatrol(), new MobSpawnerCat(), new VillageSiege(), new MobSpawnerTrader(iworlddataserver));
++            List<MobSpawner> list = ImmutableList.of(new MobSpawnerPhantom(), new MobSpawnerPatrol(), new MobSpawnerCat(), new VillageSiege(), new MobSpawnerTrader(iworlddataserver)); // Paper - This list is unused as we make these configurable in `Allow toggling special MobSpawners per world`. If this list changes, that patch likely needs updating.
+             RegistryMaterials<WorldDimension> registrymaterials = generatorsettings.d();
+             WorldDimension worlddimension = (WorldDimension) registrymaterials.a(dimensionKey);
+             DimensionManager dimensionmanager;
+diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
+index b1968103165475dd4021570e3a24deb3dc3ac796..a457696b3376348a778d3507d0dbe6514d0d94e8 100644
+--- a/src/main/java/net/minecraft/server/WorldServer.java
++++ b/src/main/java/net/minecraft/server/WorldServer.java
+@@ -231,7 +231,24 @@ public class WorldServer extends World implements GeneratorAccessSeed {
+         this.L = new ObjectLinkedOpenHashSet();
+         this.Q = flag1;
+         this.server = minecraftserver;
+-        this.mobSpawners = list;
++        // Paper start - enable/disable MobSpawners per world
++        this.mobSpawners = new java.util.ArrayList<>();
```

Comment:
What if we rather use an `ImmutableList.Builder` here as Mojang used an `ImmutableList` for the original variable?

Reply:
Ok